staging/src/k8s.io/client-go/util/workqueue/parallelizer.go

```diff
@@ -50,3 +50,36 @@ func Parallelize(workers, pieces int, doWorkPiece DoWorkPieceFunc) {
 	}
 	wg.Wait()
 }
+
+// ConditionFunc returns true if the condition is satisfied, or an error
+// if the loop should be aborted.
+type ConditionFunc func() (done bool, err error)
+
+// ParallelizeUntil is a very simple framework that allow for parallelizing
+// N independent pieces of work until the condition function return true.
+func ParallelizeUntil(workers, pieces int, doWorkPiece DoWorkPieceFunc, condition ConditionFunc) {
```

Comment:
A good idea.

Reply:
After this PR merge, I will submit a separate PR to replace all Parallelize.

Reply:
I think there was a slight misunderstanding. We should only replace the *definition* of `Parallelize` so that existing callers won't see any change. `Parallelize` would just call `ParallelizeUntil`.

I think that is a small change that should be included in this PR.